java/runtime/src/main/java/org/ray/runtime/Worker.java

```diff
@@ -95,7 +95,12 @@ public void execute(TaskSpec spec) {
         currentActorId = returnId;
       }
     } finally {
-      runtime.getWorkerContext().setCurrentTask(null, null);
+      // Don't need to reset current driver id if the worker is an actor.
+      // Because the following tasks should all have the same driver id.
+      if (!spec.isActorTask() && !spec.isActorCreationTask()) {
+        runtime.getWorkerContext().setCurrentTask(null, null);
+      }
```

Comment:
For actor tasks & actor creation tasks, we still need to reset `currentTaskId`, just don't reset `currentDriverId`. We can rename `setCurrentTask` to accept `currentTaskId` and `currentDriverId` as parameters.

Reply:
How about using 2 APIs to set the 2 fields:
`setCurrentTask()` to set the current task and class loader,
and `setCurrentDriverId()` to set the current driver id?